### PR TITLE
Move source app detection to its own middleware

### PR DESCRIPTION
**Pull request by acrollet**
## Summary

- This work is behind a feature toggle (flipper): NO

This PR moves the source app detection code from the StatsD middleware to its own middleware. The motivations for doing this are:

- Allow app controllers to use source app detection for their own purposes without hard-coding knowledge about the specific header used
  - a follow-on PR will use this to tag a specific metric with the source app
- Improve testability of the functionality
- Streamline the StatsD middleware

## Related issue(s)

- This work is in support of an OCTO initiative to improve reporting on what channel (web, mobile, etc.) a given request was initiated from (Rx Refill will be the first case)

## Testing done

- [x] *New code is covered by unit tests*
- Describe the steps required to verify your changes are working as expected:
  - add `source_app: request.env['SOURCE_APP']` to the returned object in `AdminController`
  - run `curl -H 'Source-App-Name: avs' http://127.0.0.1:3000/v0/status`
  - [ ] verify that `source_app: avs` is returned
  - run `curl -H 'Source-App-Name: foo' http://127.0.0.1:3000/v0/status`
  - [ ] verify that `source_app: not_in_allowlist` is returned
  - run `curl http://127.0.0.1:3000/v0/status`
  - [ ] verify that `source_app: not_provided` is returned

## What areas of the site does it impact?
- StatsD middleware
- Decision Review Evidences Controller

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).

### CODE REVIEW

**Review by mdewey — APPROVED** [review]
LGTM

**Review by tpharrison — APPROVED** [review]

File: lib/statsd_middleware.rb [hunk 1]
```diff
@@ -4,150 +4,6 @@ class StatsdMiddleware
   STATUS_KEY   = 'api.rack.request'
   DURATION_KEY = 'api.rack.request.duration'
 
-  MODULES_APP_NAMES = Set.new %w[
```
Comment (rmtolmach) [inline lib/statsd_middleware.rb:7]: Is there any documentation that needs to be updated to reflect these app names moving?

**Comment by rmtolmach** [comment]
**Testing**
With my rails server running locally.
```bash
rebeccatolmach ~/git/vets-api curl -H 'Source-App-Name: avs' http://127.0.0.1:3000/v0/status
{"source_app":"avs","git_revision":"MISSING_GIT_REVISION","db_url":null,"postgres_up":true,"redis_up":true,"redis_details":{"app_data_redis":true,"rails_cache":true,"sidekiq_redis":true}}
rebeccatolmach ~/git/vets-api curl -H 'Source-App-Name: foo' http://127.0.0.1:3000/v0/status
{"source_app":"not_in_allowlist","git_revision":"MISSING_GIT_REVISION","db_url":null,"postgres_up":true,"redis_up":true,"redis_details":{"app_data_redis":true,"rails_cache":true,"sidekiq_redis":true}}
rebeccatolmach ~/git/vets-api curl http://127.0.0.1:3000/v0/status
{"source_app":"not_provided","git_revision":"MISSING_GIT_REVISION","db_url":null,"postgres_up":true,"redis_up":true,"redis_details":{"app_data_redis":true,"rails_cache":true,"sidekiq_redis":true}}
```
✅ all testing steps look good.

**Review by shaunburdick — APPROVED** [review]
`pr.env['LGTM']`

File: lib/statsd_middleware.rb (hunk 1 above)
Comment (acrollet) [reply lib/statsd_middleware.rb:7]: good point - I don't see any of the constants referenced anywhere else in the repo, but step #4 on the [new FE app docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/creating-a-new-application) will need to be updated.

Comment (acrollet) [reply lib/statsd_middleware.rb:7]: (which I'll be happy to take care of)

Comment (rmtolmach) [reply lib/statsd_middleware.rb:7]: ok, thank you! appreciate it.

**Review by rmtolmach — APPROVED** [review]

File: lib/statsd_middleware.rb (hunk 1 above)
Comment (acrollet) [reply lib/statsd_middleware.rb:7]: linking docs issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/89767

**Pull request merged**